modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/operation/ComponentMessageProcessor.java

```diff
@@ -230,12 +231,22 @@ public CoreEvent process(CoreEvent event) throws MuleException {
   private CoreEvent addContextToEvent(CoreEvent event, Context ctx) {
     final Optional<Scheduler> currentScheduler = ctx.getOrEmpty(PROCESSOR_SCHEDULER_CONTEXT_KEY);
 
-    if (ctx.hasKey(POLICY_NEXT_OPERATION)) {
+    if (ctx.hasKey(POLICY_NEXT_OPERATION) || ctx.hasKey(POLICY_IS_PROPAGATE_MESSAGE_TRANSFORMATIONS)) {
+      Function<Context, Context> context = innerChainCtx -> {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Refactor this code to use the more specialised Functional Interface 'UnaryOperator<Context>' ([squid:S4276](https://sonarqube.kbuild.msap.io/coding_rules#rule_key=squid%3AS4276))

[See it in SonarQube](https://sonarqube.kbuild.msap.io/project/issues?id=mule-runtime-mule&pullRequest=7705&issues=AWkM1PiOotIXnmsU5TvI&open=AWkM1PiOotIXnmsU5TvI)

[ISSUE_KEY]: AWkM1PiOotIXnmsU5TvI